can/typechecking.py

```diff
@@ -18,3 +23,18 @@
 
 # Used for the Abstract Base Class
 Channel = typing.Union[int, str]
+
+BusConfig = mypy_extensions.TypedDict(
```

Comment:
So not really sure what the best way to go about this is.. On one hand, you get some additional type inference.

However, the problem with using a `TypedDict` is that it doesn't allow for optional types. `total=False` only allows omitting keys, and does not allow extra keys, so if various backends introduce their own config keys, they'd need to be specified here. This could potentially get really long.

Reply:
I'm unsure about this too... The thing is that the "dispatcher class" Bus is else throwing the type information of the specific interfaces away right?

Reply:
Requiring backends to add all optional config keys here seems to be a recipe for disaster. 

Annotating as a `NewType("BusConfig", dict)` might be a good comprise? It may offer a small advantage over a simple type alias to `Dict[str, any]` - only `dicts` created using a function annotated as returning a `typechecking.BusConfig` would pass mypy.